File: ts-sdk/src/sdk/models/shared/createwalletrequest.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class CreateWalletRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name: string;
}

File: python-sdk/src/formanceapi/models/shared/createwalletrequest.py
import dataclasses
from dataclasses_json import dataclass_json
from formanceapi import utils
from typing import Any, Optional


@dataclass_json
@dataclasses.dataclass
class CreateWalletRequest:
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metadata') }})
    